libs/block-editor/src/lib/extensions/dragHandler.extention.ts

```diff
@@ -0,0 +1,156 @@
+            let pos = blockPosAtCoords(coords, view);
+            if (pos != null) {
+                view.dispatch(
+                    view.state.tr.setSelection(NodeSelection.create(view.state.doc, pos))
+                );
+                let slice = view.state.selection.content();
+                e.dataTransfer.clearData();
+                debugger;
+                e.dataTransfer.setDragImage(nodeToBeDragged, 10, 10);
+                view.dragging = { slice, move: true };
+            }
+        }
+
+        // Get the direct child of the Editor.
+        function getDirectChild(node) {
+            while (node && node.parentNode) {
+                if (
+                    node.classList?.contains('ProseMirror') ||
+                    node.parentNode.classList?.contains('ProseMirror')
+                ) {
+                    break;
+                }
+                node = node.parentNode;
+            }
+            return node;
+        }
+
+        // Check if not has content.
+        function nodeHasContent(view: EditorView, inside: number): boolean {
+            return !!view.nodeDOM(inside)?.textContent;
+        }
+
+        function createDragHandler(editorView: EditorView) {
+            dragHandler = document.createElement('span');
+            dragHandler.setAttribute('draggable', 'true');
+            dragHandler.textContent = 'drag_indicator';
+            dragHandler.className = 'material-icons';
+            dragHandler.style.position = 'absolute';
+            dragHandler.style.cursor = 'grab';
+            dragHandler.addEventListener('dragstart', (e) => dragStart(e, editorView));
+            document.body.appendChild(dragHandler);
+        }
+
+        return [
+            new Plugin({
+                key: new PluginKey('dragHandler'),
+                view: (editorView) => {
+                    createDragHandler(editorView);
+                    return {
+                        update(view, prevState) {},
+                        destroy() {
+                            removeNode(dragHandler);
+                            dragHandler = null;
+                        }
+                    };
+                },
+                props: {
+                    handleDOMEvents: {
+                        drop(view, event) {
+                            setTimeout(() => {
+                                debugger;
+                                let node = document.querySelector('.ProseMirror-hideselection');
+                                if (node) {
+                                    node.classList.remove('ProseMirror-hideselection');
+                                }
+                            }, 50);
+                            event.stopPropagation();
+                            return false;
+                        },
+                        mousemove(view, event) {
+                            let coords = { left: event.clientX + 50, top: event.clientY };
```

Comment:
What is `+ 50`?

Reply:
to consider the gap between the handler element, and  block element that we will drag